src/ess/bifrost/live.py

```diff
@@ -0,0 +1,95 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2025 Scipp contributors (https://github.com/scipp)
+
+"""Live data reduction workflows for BIFROST."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import NewType
+
+import sciline
+import scipp as sc
+import scippnexus as snx
+
+from ess.reduce.streaming import EternalAccumulator, StreamProcessor
+from ess.spectroscopy.types import (
+    EnergyData,
+    InstrumentAngles,
+    NeXusData,
+    NeXusDetectorName,
+    RunType,
+    SampleRun,
+)
+
+from .workflow import BifrostWorkflow
+
+
+@dataclass(frozen=True, kw_only=True, slots=True)
+class CutAxis:
+    """Axis and binds for cutting 4D Q-E data."""
```

Comment:
Is this already more advanced than a simple (|Q|, dE), i.e., for arbitrary planes?

Reply:
If you can write a Pydantic model for it, yes. The set of supported field types is limited, but could be extended: https://github.com/scipp/beamlime/blob/b40409fcb57604c7ee3bc82e989f86a8a046339c/src/ess/livedata/dashboard/widgets/param_widget.py#L70-L118.

There are also pre-made models for various "bin edges" that should be reused.